dev/add_common_fields_strelka2.py

```diff
@@ -0,0 +1,149 @@
+            return "0/1"
+    def alleles_to_gt(val):
+        gt_indices = {gt.upper(): i for i, gt in enumerate([record.ref] + list(record.alts))}
+        tumor_gts = [gt_indices[x.upper()] for x in val if x in gt_indices]
+        if tumor_gts and val not in known_names:
+            if max(tumor_gts) == 0:
+                tumor_gt = "0/0"
+            elif 0 in tumor_gts:
+                tumor_gt = "0/%s" % min([x for x in tumor_gts if x > 0])
+            else:
+                tumor_gt = "%s/%s" % (min(tumor_gts), max(tumor_gts))
+        else:
+            tumor_gt = name_to_gt(val)
+        return tumor_gt
+    nt_val = record.info['NT']
+    normal_gt = name_to_gt(nt_val)
+    sgt_val = record.info['SGT'].split("->")[-1]
+    if not sgt_val:
+        tumor_gt = "0/0"
+    else:
+        tumor_gt = alleles_to_gt(sgt_val)
+    # convert to tuple for use in pysam
+    tumor_gt = tuple(map(int, tumor_gt.split("/")))
+    normal_gt = tuple(map(int,normal_gt.split ("/")))
+    return tumor_gt, normal_gt
+
+def _calc_AD(record):
+    """From strelka2 docs:
+    SNPs:
+    refCounts = Value of FORMAT column $REF + “U” (e.g. if REF="A" then use the value in FOMRAT/AU)
+    altCounts = Value of FORMAT column $ALT + “U” (e.g. if ALT="T" then use the value in FOMRAT/TU)
+    tier1RefCounts = First comma-delimited value from $refCounts
+    tier1AltCounts = First comma-delimited value from $altCounts
+    Indels:
+    tier1RefCounts = First comma-delimited value from FORMAT/TAR
+    tier1AltCounts = First comma-delimited value from FORMAT/TIR
+    """
+    tum_alt_ct = 0
+    tum_ref_ct = 0
+    norm_alt_ct = 0
+    norm_ref_ct = 0
+    # only indels should have TIR field
+    if 'TIR' in record.samples[samp_list[tum_idx]]:
+        tum_ref_ct = record.samples[samp_list[tum_idx]]['TAR'][0]
+        tum_alt_ct = record.samples[samp_list[tum_idx]]['TIR'][0]
+        norm_ref_ct = record.samples[samp_list[norm_idx]]['TAR'][0]
+        norm_alt_ct = record.samples[samp_list[norm_idx]]['TIR'][0]
+
+    else:
+        snp_ref_key = record.ref + "U"
+        snp_alt_key = record.alts[0] + "U"
+        tum_ref_ct = record.samples[samp_list[tum_idx]][snp_ref_key][0]
+        tum_alt_ct = record.samples[samp_list[tum_idx]][snp_alt_key][0]
+        norm_ref_ct = record.samples[samp_list[norm_idx]][snp_ref_key][0]
+        norm_alt_ct = record.samples[samp_list[norm_idx]][snp_alt_key][0]
+    tum_ad = (tum_ref_ct, tum_alt_ct)
+    norm_ad = (norm_ref_ct, norm_alt_ct)
+    # return as tuple for pysam compatibility
+    return tum_ad, norm_ad
+
+
+
+def _calc_AF_AD(record):
+    """From strelka2 docs:
+    SNPs:
+    Somatic allele freqeuncy is $tier1AltCounts / ($tier1AltCounts + $tier1RefCounts)
+    Indels:
+    Somatic allele freqeuncy is $tier1AltCounts / ($tier1AltCounts + $tier1RefCounts)
+    """
+    tum_af = 0.0
+    norm_af = 0.0
+    tum_ad, norm_ad = _calc_AD(record)
+    try:
+        tum_af = tum_ad[1]/sum(list(tum_ad))
```

Comment:
```
chr16	24438703	.	AGATGGATG	A	.	PASS	IC=12;IHP=2;MQ=60.00;MQ0=0;NT=ref;QSI=1;QSI_NT=1;RC=14;RU=GATG;SGT=het->het;SOMATIC;SomaticEVS=9.19;TQSI=1;TQSI_NT=1	BCN50:DP:DP2:DP50:FDP50:SUBDP50:TAR:TIR:TOR	0.00:18:18:12.11:0.22:0.00:1,1:10,10:8,10	0.00:2:2:1.84:0.00:0.00:0,0:0,0:2,2
```
the above record (deletion) breaks here because the TIR+TAR is zero and a ZeroDivisionError is triggered. you can add an additional except block to handle that specific case:
```python
try:
    calculate af
except ZeroDivisionError:
    af = 0
except Exception:
    ...
```